Release

jyecusch opened the pull request.

asalkeld committed:
feat(gcp): Add gcp provider

asalkeld committed:
fix(target): Parsing of extra targets

asalkeld committed:
fix: Improve deployment output

asalkeld committed:
fix(pulumi): Produce a better message when pulumi is not installed

tjholm committed:
chore: Update README.

jyecusch committed:
Merge pull request #128 from nitrictech/feature/update-readme

Update README

tjholm committed:
Merge pull request #130 from asalkeld/target-parsing

fix(target): Parsing of extra targets

davemooreuws committed:
ci: release only on prod

tjholm committed:
Merge pull request #132 from nitrictech/ci/release-only-prod

ci: release only on prod

tjholm committed:
chore: Simplify busy updates for pulumi deployments.

asalkeld committed:
Merge pull request #133 from nitrictech/feature/simplify-pulumi-updates

Simplify busy updates for pulumi deployments

jyecusch committed:
fix: fix unable to put object in S3 bucket

jyecusch committed:
Merge pull request #134 from nitrictech/fix/s3-buckets-put

fix: fix unable to put object in S3 bucket

tjholm reviewed (commented):
Do we want to release the new membrane version and set this as the new default before releasing this version?

jyecusch committed:
fix: invalid s3 action

the action would be `s3:ListBucket` not `s3:ListBuckets` and we don't need it right now before we have a list feature.

jyecusch committed:
Merge pull request #136 from nitrictech/fix/s3-policies

fix: invalid s3 action

jyecusch commented:
> Do we want to release the new membrane version and set this as the new default before releasing this version?

Good call, I've moved this to draft until the Membrane release is out

jyecusch committed:
Merge pull request #127 from asalkeld/pulumi-missing

fix(pulumi): Produce a better message when pulumi is not installed

asalkeld committed:
feat: rename commands to be more user friendly

1. remove "stack describe"
2. rename "stack new" to "new"
3. rename "deployment apply" to "stack up"
4. rename "deployment delete" to "stack down"
5. rename "deployment list" to "stack list"
6. remove user aliases

asalkeld committed:
chore: remove unimplemented command "provider list"

asalkeld committed:
feat: Add aliases for up,down and list

asalkeld committed:
Merge pull request #140 from asalkeld/dep-to-stack

Rename deployment command to stack

asalkeld committed:
refactor: rename Stack to Project

After the commands have changed, update the internal types to follow.

raksiv committed:
wip: skeleton for new CLI Readme

asalkeld committed:
feat: Improve the help

asalkeld committed:
feat: Extend "nitric new" to include the handler glob and write to nitric.yaml

fixes #139

tjholm committed:
feat: Add stack name to functions in aws deployments.

tjholm committed:
feat: Add lambda permission for resource discovery.

asalkeld committed:
chore: delete "target list"

asalkeld committed:
feat: move target to stack and implement "stack new"

tjholm committed:
Merge pull request #84 from asalkeld/gcp

feat: Implement gcp provider

asalkeld committed:
fix(help): Bring help text inline with issue 142

fixes #142

asalkeld committed:
chore: remove unused e2e script

jyecusch committed:
chore: remove dot

tjholm committed:
Merge pull request #154 from asalkeld/combined-new-commands

PR combining all new commands

asalkeld committed:
feat(run): capture codeAsConfig container logs for better errors

fixes #98

tjholm committed:
Merge pull request #156 from asalkeld/better-output-when-run-fails

feat(run): capture codeAsConfig container logs for better errors

asalkeld committed:
fix: Prevent codeAsConfig containers from clashing with running containers

- name the containers a bit better to prevent scoping issues
- start the LocalServices earlier so we can check Running() and fail earlier

fixes #149

asalkeld committed:
fix(nitric new): Allow a non-interactive command

fixes #96

asalkeld committed:
fix: Provide suggestion when project file is missing

fixes #145

asalkeld committed:
feat: Add a confirmation to the down command

closes #151

asalkeld committed:
Merge pull request #158 from asalkeld/stack-create-with-arg

fix(nitric new): Allow a non-interactive command

asalkeld committed:
Merge pull request #159 from asalkeld/missing-proj-file

fix: Provide suggestion when project file is missing

asalkeld committed:
chore: Remove stackName as it is in the new stack config file

- remove some dead code
- add a unit test

asalkeld committed:
chore: rename provider functions to match commands

Apply -> Up
Delete -> Down

asalkeld committed:
refactor: Rename provider member variables to better match their names

tjholm committed:
fix(azure): Don't deploy event subscriptions within the scope of an apply.

asalkeld committed:
fix: create the error in codeAsConfig based only on done.StatusCode

tjholm committed:
Merge pull request #164 from asalkeld/code-as-config-log-output

fix: create the error in codeAsConfig based only on done.StatusCode

HomelessDinosaur committed:
Merge pull request #162 from asalkeld/remove-stack-name-flag

cleanup: Remove stack name flag

asalkeld committed:
docs: auto generate the command help section of the readme

closes nitrictech/nitric-website#320

jyecusch committed:
chore: handle err from NewEventSubscription

tjholm committed:
Merge pull request #163 from nitrictech/fix/azure-sub-deployments

Don't deploy event subscriptions inside an apply

tjholm committed:
refactor: Update azure resources to use native provider.

tjholm committed:
chore: cleanup.

asalkeld committed:
feat: Catch panics and print a issue link for users to report bugs

closes #146

asalkeld committed:
Merge pull request #160 from asalkeld/prompt-down

feat: Add a confirmation to the down command

asalkeld committed:
Merge pull request #157 from asalkeld/check-if-running

fix: Prevent codeAsConfig containers from clashing with running containers

asalkeld committed:
Merge pull request #166 from asalkeld/feat/refresh-readme

docs: refresh readme

asalkeld committed:
feat: Add a new command for feedback

closes #155

asalkeld committed:
ci: build and test on ubuntu, windows and mac

asalkeld committed:
fix: build target

asalkeld committed:
fix: merge build and build-windows makefile targets

This is so we can have the same ci build target

davemooreuws committed:
Merge pull request #174 from asalkeld/fix-build-target

fix: build target

asalkeld committed:
ci: on windows only build

jyecusch committed:
fix: update default membrane version

jyecusch committed:
fix: apply correct s3 list action

jyecusch committed:
Merge pull request #176 from nitrictech/fix/update-membrane-default

fix: update default membrane version

davemooreuws committed:
Merge pull request #173 from asalkeld/multi-os-ci-test

ci: build and test on ubuntu, windows and mac

tjholm committed:
chore: Update to compatible membrane version.

asalkeld committed:
refactor: make the javascript and typescript files easier to diff

asalkeld committed:
fix: Make sure docker image names are lowercase

closes #172

asalkeld committed:
fix: Make sure that Ignores make their way to docker build


The problem is that boxygen Ignore has no effect. So publish
a new function "BuildIgnore" on the runtime interface which
is used by the build to pass to the docker excludes.

davemooreuws committed:
Merge pull request #177 from nitrictech/fix/s3-list-action

fix: apply correct s3 list action

asalkeld committed:
chore: remove dead code

asalkeld committed:
test: more runtime unit tests

asalkeld committed:
Merge pull request #178 from asalkeld/exclude-node_modules

Fix: docker build error "repository name must be lowercase "

HomelessDinosaur committed:
Merge pull request #169 from asalkeld/catch-panics

feat: Catch panics and print a issue link for users to report bugs

tjholm committed:
Merge pull request #181 from asalkeld/dead-code-and-more-tests

Remove dead code and add more tests

asalkeld committed:
feat: Add a new command "info" for gathering  information about Nitric and it's environoment

closes #171

tjholm committed:
Merge pull request #153 from nitrictech/feat/aws-stack-env

AWS Resource Discovery

raksiv committed:
chore: don't show apis,schedules or topics if they aren't defined.

asalkeld committed:
docs: Add issue templates for bug-report and enhancement

part of nitrictech/nitric#248

tjholm committed:
Merge pull request #182 from nitrictech/fix/run-area-refresh

chore: only show subs,topics,apis resources if they are in use

tjholm committed:
feat: Add resource group name to deployed container apps.

Limit scope of resource queries to sibling stack resources.

tjholm committed:
chore: run fmt.

tjholm committed:
chore: Apply tag contributor access to resource resolution.

asalkeld committed:
fix: go install of CompileDaemon

New versions of golang don't like "go get" to install

tjholm committed:
fix(azure): Ensure correct imageUris are given when pushing container images.

tjholm committed:
fix: Set MIN_WORKERS, based on registered function workers.

tjholm committed:
Merge pull request #184 from nitrictech/fix/min-workers

Set MIN_WORKERS, based on registered function workers.

tjholm committed:
fix: allow abbreviations to be optional.

tjholm committed:
chore: Add missing resource group name for storage containers and queues.

asalkeld committed:
Merge pull request #170 from asalkeld/nitric-feedback

feat: add a new command for feedback

asalkeld committed:
Merge pull request #183 from asalkeld/info-cmd-for-issues

Info cmd for issues and add templates

tjholm committed:
Merge pull request #185 from asalkeld/fix-go-install

fix: go install of CompileDaemon

tjholm committed:
chore: bump membrane version.

jyecusch committed:
Merge pull request #167 from nitrictech/feat/azure-refator

Azure deployment refactor
